### PR TITLE
Update golang version to 1.17

### PR DESCRIPTION
**Pull request by praveenkumar**

### CODE REVIEW

File: .github/workflows/publish-docs.yml [hunk 1]
```diff
@@ -13,6 +13,8 @@ jobs:
       matrix:
         os:
           - ubuntu-latest
+        go:
+          - 1.17
```
Comment (cfergeau) [inline .github/workflows/publish-docs.yml:17]: Is this using go?

File: images/build-e2e/Dockerfile [hunk 2]
```diff
@@ -1,4 +1,4 @@
-FROM registry.access.redhat.com/ubi8/go-toolset:1.16.7 as builder
+FROM registry.access.redhat.com/ubi8/go-toolset:1.17.10 as builder
```
Comment (cfergeau) [inline images/build-e2e/Dockerfile:1]: Only go-toolset:1.17.7 is available.

File: images/build-integration/Dockerfile [hunk 3]
```diff
@@ -1,4 +1,4 @@
-FROM registry.access.redhat.com/ubi8/go-toolset:1.16.7 as builder
+FROM registry.access.redhat.com/ubi8/go-toolset:1.17.10 as builder
```
Comment (cfergeau) [inline images/build-integration/Dockerfile:1]: Same here

File: .github/workflows/publish-docs.yml (hunk 1 above)
Comment (praveenkumar) [reply .github/workflows/publish-docs.yml:17]: I used the `update-go-version.sh` and looks like https://github.com/code-ready/crc/blob/main/update-go-version.sh#L18-L20 is culprit to adding it.

File: images/build-e2e/Dockerfile (hunk 2 above)
Comment (praveenkumar) [reply images/build-e2e/Dockerfile:1]: Yeah it is again https://github.com/code-ready/crc/blob/main/update-go-version.sh#L15-L16, looks like after executing this script we have to manually check or if there is way to get latest toolset version from registry.

Comment (cfergeau) [reply images/build-e2e/Dockerfile]: Yeah, it had the same issue with 1.16, the version needed to be manually adjusted.

Comment (praveenkumar) [reply images/build-e2e/Dockerfile]: @cfergeau Updated.

File: .github/workflows/publish-docs.yml (hunk 1 above)
Comment (praveenkumar) [reply .github/workflows/publish-docs.yml]: updated

File: images/build-integration/Dockerfile (hunk 3 above)
Comment (praveenkumar) [reply images/build-integration/Dockerfile]: updated manually

**Review by cfergeau — APPROVED** [review]

**Pull request merged**